maintainers-guide.md

```diff
@@ -168,27 +168,35 @@ organization](https://github.com/bazelbuild) to be able to add or remove labels.
-  breaking change that was not compliant to the [Breaking
+* [**P0**](https://github.com/bazelbuild/bazel/labels/P0) - Major broken
+  functionality that causes a Bazel release (minus release candidates) to be
+  unusable, or a downed service that severely impacts development of the Bazel
+  project. This includes regressions introduced in a new release that blocks a
+  significant number of users, or an incompatible breaking change that was not
+  compliant to the [Breaking
   Change](https://docs.google.com/document/d/1q5GGRxKrF_mnwtaPKI487P8OdDRh2nN7jX6U-FXnHL0/edit?pli=1#heading=h.ceof6vpkb3ik)
   policy. No practical workaround exists.
-* **P1** - Critical defect or feature which should be addressed in the next
-  release, or a serious issue that impacts many users (including the development
-  of the Bazel project), but a practical workaround exists. Typically does not
-  require immediate action. In high demand and planned in the current quarter's
-  roadmap.
-* **P2** - Defect or feature that is tied to roadmaps or upcoming releases.
-  Moderate live issue in a released Bazel version that is inconvenient for a
-  user that needs to be addressed in an upcoming release and/or a easy workaround
-  exists.
-* **P3** - Desirable minor bug fix or enhancement with small impact. Not
-  prioritized into Bazel roadmaps or any imminent release. May never be fixed.
-* **P4** - Low priority defect or feature request that is unlikely to get
-  closed. Can also be kept open for a potential re-prioritization if more users
-  are impacted.
+* [**P1**](https://github.com/bazelbuild/bazel/labels/P1) - Critical defect or
+  feature which should be addressed in the next release, or a serious issue that
+  impacts many users (including the development of the Bazel project), but a
+  practical workaround exists. Typically does not require immediate action. In
+  high demand and planned in the current quarter's roadmap.
+* [**P2**](https://github.com/bazelbuild/bazel/labels/P2) - Defect or feature
+  that is tied to roadmaps or upcoming releases.  Moderate live issue in a
+  released Bazel version that is inconvenient for a user that needs to be
+  addressed in an upcoming release and/or a easy workaround exists.
+* [**P3**](https://github.com/bazelbuild/bazel/labels/P3) - Desirable minor bug
+  fix or enhancement with small impact. Not prioritized into Bazel roadmaps or
+  any imminent release. May never be fixed.
+* [**P4**](https://github.com/bazelbuild/bazel/labels/P4) - Low priority defect
+  or feature request that is unlikely to get closed. Can also be kept open for a
+  potential re-prioritization if more users are impacted.
+* [**ice-box**](https://github.com/bazelbuild/bazel/issues?q=label%3Aice-box+is%3Aclosed)
+  - Issues that we currently don't have time to deal with nor the
+  time to accept contributions. We will close these issues to indicate that
+  nobody is working on them, but will continue to monitor their validity over
```

Comment:
> We will close these issues to indicate that nobody is working on them

Most of the open issues are not being worked on, right now.

If we monitor them, why not keep them open? Closing issues means that duplicates are more likely to be filed.

Reply:
It's a stronger signal (see also the second half of the sentence). Also it doesn't make our bugs grow unbounded. I doubt that duplicates are a big issue but if they are we can reconsider this new label.

Reply:
IMO because these issues will never be closed otherwise, so our bug tracker will grow unbounded. I think an ever-increasing growth of bugs reflects badly on the project, and the more issues there are, the harder it becomes to see the signal from the noise (for us and for everyone not familiar with the conventions we follow).

(Not feeling too strong about that though. I'll just filter out icebox issues from any queries I care about.)

Reply:
+1 closing them does not serve a purpose. If they are good ideas but not a priority and unstaffed we want them easily visible so they can be upvoted.

Reply:
In practice, though, the icebox will be a dump ground for bugs. I don't see this en-masse up-voting happening. We don't have the volume of vscode.

I'd much rather deal with the occasional duplicate against already-existing iceboxed issues, and closing them as such when they come in.

Reply:
I'm for closing. We will never be in a position where we think: "Hey, I know what - I'm so bored, I'll look through some P4 or Icebox issues for something to do!"

IMHO it's much more important to keep the issue tracker tidy and the count going down instead of up.

Reply:
@aiuto @laurentlb Are you willing to give this a try?

Reply:
Can someone propose a list of issues where we'll label with `ice-box`? I think it'll be easier to consider the optics of closing an issue with the `ice-box` label if we have a concrete list to work with.

Reply:
Jin, there are already some issues that use this label: https://github.com/bazelbuild/bazel/issues?q=label%3Aice-box+is%3Aclosed

Reply:
Can we please consider the optics of *closing* issues without resolution or explanation? See: https://github.com/bazelbuild/bazel/issues/8177#issuecomment-628850087

As an issue author, I don't have insight into how the project's labelling system works. If the maintainer closes my seemingly valid bug report or FR (which I may have put a lot of work into filing) and puts it into an ice-box because they want the issue out-of-sight and hopes that eventually someone will pick it up, it will dis-incentivize me to file new issues. I would very much rather see my issue acknowledge and staying open (while collecting reactions and responses), than to have it closed and put away.

Reply:
Thanks Jin for calling this out, I should not have closed that without further comments.

To your general point: As an issue author, I would rather like to see an issue being closed / put in the ice-box explicitly than having it hang around for a long time without response.

Reply:
Chiming in in favor of closing (but with explanation).

Issues are ultimately just organizational helpers. Their exact state, status, and format isn't what's most important. What's most important is the understanding they facilitate between people.

Closing doesn't make them disappear (okay it sort of does, but that can be worked with). It's just another signal. It's up to us what understanding we want to attach to that signal. One option is "this is not worth our time and we'll never look at it again". But it's not the only option and I think the larger open-source community gets that. The conversation doesn't have to stop just because an issue is closed.

As long as our policy is clear and anyone who closes a bug adds a comment clearly explaining why, including acknowledgement of the value of the issue and an offer to re-open / re-prioritize under well-clarified circumstances, I think we're in great shape. In this case closing offers a more decisive outcome than a forever-open bug that's really closed in spirit. If communicated well I think this is a more fair response to stakeholder and they can appreciate that.

There's no shortage of GitHub issues in the open-source world that stay open forever, at whatever priority and with whatever tags, where the last 20 comments are "any chance of doing this?", "is this still alive?", "will this ever be fixed?". Some of these get tweeted because they look so absurd. It leaves people in an under-communicated limbo that they can build no expectations around. And they clutter up the open issues list so no one can tell what open issues really matter or get actual attention. Labels *may* help with that. But they're just another signal, with their own organizational drawbacks too.

Reply:
Agree with @gregestren . It is important we close issues, but when we do so, that we say why. Canned responses can help, as well as a link to the document (this?) that describes our bug process and what each label means.

One more reason for closing: filing bugs costs *nothing*. Anyone can trivially file a feature request that sounds reasonable but that can take years for us to build. I don't want our bug tracker to be hostage to such issues, so I want the ability to close them (again, with a rationale, and with the option of revisiting those decisions).

Reply:
Does anyone know how other Open Source project handle this? I only found a statement from owncloud so far: https://gist.github.com/jancborchardt/6155185#clean-up-inactive-issues

Reply:
Works for me.

Reply:
Thanks Laurent.

Two more data points:
- Envoy uses a stale bot to close issues that are not tagged with a special label after ~1 month: https://github.com/envoyproxy/envoy/blob/master/.github/stale.yml
- Kubernetes: https://kubernetes.io/docs/contribute/review/for-approvers/#issue-lifecycle-labels

Reply:
Just to clarify: my main concern was to another label (we have lots), that means "almost like P4, but not quite". But yes, that's a minor concern and I'm fine with proceeding with the change.

I would also be fine with auto-closing P4 issues after some time of inactivity, if we want to keep the number of open issues lower.

Reply:
Explicitly mentioned comments/reactions after our offline discussion.